### PR TITLE
Duplicate check on importing in find unlinked files

### PR DESCRIPTION
**Pull request by R-Zhou4**
This PR resolves the issue of duplicate entries being imported when clicking the "Import" button multiple times by refining the import process.
"Closes https://github.com/koppor/jabref/issues/551".

### Key Changes:
**UnlinkedFilesDialogViewModel Class**:

- **edit startImport Method**:
  - **Import Logic**: Uses the `ImportHandler.getEntriesToImport` method to retrieve a list of entries.
  - **Deduplication Logic**: Prevents duplicate entries from being imported by using the `importHandler.findDuplicate` method.

- **add getFilePathFromEntry Method**:
  - Extracts the correct file path from `BibEntry`.
  - Returns a valid path, or an empty path if no valid fields are found.

**ImportHandler Class**:

- **add getEntriesToImport Method**:
  - Processes the incoming list of files and extracts importable `BibEntry` objects.
  - Handles different file types (e.g., PDF and BibTeX) and returns unique entries.
  - Maintains import integrity by creating empty entries for unsupported file types.

**BibEntry Class**:

- **add getCiteKey Method**:
  - Adds a method for retrieving the citation key to support duplicate entry detection.

### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/model/entry/BibEntry.java [hunk 1]
```diff
@@ -1052,6 +1052,10 @@ public KeywordList getFieldAsKeywords(Field field, Character keywordSeparator) {
         return keywords;
     }
 
+    public Optional<String> getCiteKey() {
```
Comment (koppor) [inline src/main/java/org/jabref/model/entry/BibEntry.java:1055]: Remove. Use https://github.com/JabRef/jabref/blob/4ce83c49c69fc7c14b1507974b2540e941dff26c/src/main/java/org/jabref/model/entry/BibEntry.java#L403

Move method below up, too.

**Review by koppor — COMMENTED** [review]
Small initial comment while on the road.

File: src/main/java/org/jabref/gui/externalfiles/UnlinkedFilesDialogViewModel.java [hunk 2]
```diff
@@ -166,10 +188,49 @@ public void startImport() {
                                                      progressTextProperty.unbind();
                                                      taskActiveProperty.setValue(false);
                                                  })
-                                                 .onSuccess(resultList::addAll);
+                                                 .onSuccess(new Consumer<List<ImportFilesResultItemViewModel>>() {
+                                                     @Override
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/externalfiles/UnlinkedFilesDialogViewModel.java:192]: You can use a lambda here directly no need for anonymous method, Intellij should be able to do this as well  https://www.jetbrains.com/help/rider/ConvertToLambdaExpression.html

Comment (R-Zhou4) [reply src/main/java/org/jabref/gui/externalfiles/UnlinkedFilesDialogViewModel.java]: Replaced with a lambda now

File: src/main/java/org/jabref/model/entry/BibEntry.java (hunk 1 above)
Comment (R-Zhou4) [reply src/main/java/org/jabref/model/entry/BibEntry.java]: Updated: getCiteKey() has been removed.

**Review by koppor — CHANGES_REQUESTED** [review]
Unfortunately, this is implemented way too compilcated.

The PDF file is NOT changed if the dialog is opended. Thus, one "just" needs to query the library if any file is attached to the entry.

Use `entry.getFiles()`, convert each path to an absolute one and check for the file.

You could also store in the dialog the set of imported paths and just check for equality when another path is requested to be imported.

**Pull request closed without merging**